File: lib/ts/src/internal/schema/CreateMockMsgPactSchemaFromFileJsonMap.ts
import { MockMsgPactSchema } from '../../MockMsgPactSchema';
import { createMsgPactSchemaFromFileJsonMap } from './CreateMsgPactSchemaFromFileJsonMap';
import { getMockMsgPactJson } from './GetMockMsgPactJson';

export function createMockMsgPactSchemaFromFileJsonMap(jsonDocuments: Record<string, string>): MockMsgPactSchema {
    const finalJsonDocuments = { ...jsonDocuments };
    finalJsonDocuments['mock_'] = getMockMsgPactJson();

    const msgPactSchema = createMsgPactSchemaFromFileJsonMap(finalJsonDocuments);

    return new MockMsgPactSchema(
        msgPactSchema.original,
        msgPactSchema.full,
        msgPactSchema.parsed,
        msgPactSchema.parsedRequestHeaders,
        msgPactSchema.parsedResponseHeaders,
    );
}

File: lib/py/msgpact/internal/schema/CreateMockMsgPactSchemaFromFileJsonMap.py
from typing import TYPE_CHECKING
from ...MockMsgPactSchema import MockMsgPactSchema


def create_mock_vers_api_schema_from_file_json_map(json_documents: dict[str, str]) -> 'MockMsgPactSchema':
    from .GetMockMsgPactJson import get_mock_vers_api_json
    from .CreateMsgPactSchemaFromFileJsonMap import create_vers_api_schema_from_file_json_map

    final_json_documents = json_documents.copy()
    final_json_documents["mock_"] = get_mock_vers_api_json()

    msgpact_schema = create_vers_api_schema_from_file_json_map(
        final_json_documents)

    return MockMsgPactSchema(msgpact_schema.original, msgpact_schema.parsed, msgpact_schema.parsed_request_headers, msgpact_schema.parsed_response_headers)
